src/circuit_builder.rs

```diff
@@ -90,215 +443,138 @@ pub enum CircuitErrors {
-            pub_params: &PublicParameters,
-            verifier_key: &VerifierKey,
-            transcript_initialisation: &'static [u8],
-            proof: &Proof,
-            pub_inputs: &[PublicInput],
-        ) -> Result<()> {
-            let (_, vk) = pub_params.trim(1 << 9)?;
-            // New Verifier instance
-            let mut verifier = Verifier::new(transcript_initialisation);
-            // Fill witnesses for Verifier
-            self.gadget(verifier.mut_cs())?;
-            verifier.verifier_key = Some(*verifier_key);
-            verifier.verify(proof, &vk, &self.build_pi(pub_inputs)?)
-        }
-    }
+    );
 
     #[test]
     fn test_full() -> Result<()> {
         // Generate CRS
         let pub_params = PublicParameters::setup(1 << 10, &mut rand::thread_rng())?;
-        // Generate circuit compilation params
-        let inputs = [
-            BlsScalar::from(25u64),
-            BlsScalar::from(5u64),
-            BlsScalar::from(30u64),
-            BlsScalar::from(125u64),
-        ];
 
         // Initialize the circuit
         let mut circuit = TestCircuit::default();
-        circuit.inputs = Some(&inputs);
         {
             // Compile the circuit
-            let (prover_key, verifier_key, _) = circuit.compile(&pub_params)?;
+            circuit.compile(&pub_params)?;
+
+            let commit_key = circuit.commit_key().as_ref().cloned().unwrap();
+            let opening_key = circuit.opening_key().as_ref().cloned().unwrap();
+            let prover_key = circuit.prover_key().as_ref().cloned().unwrap();
+            let verifier_key = circuit.verifier_key().as_ref().cloned().unwrap();
+
             // Write the keys
             use std::fs::File;
             use std::io::Write;
+
+            let mut commit_file = File::create("ck_testcirc")?;
+            commit_file.write(commit_key.to_bytes()[..].as_ref())?;
+
+            let mut opening_file = File::create("ok_testcirc")?;
+            opening_file.write(opening_key.to_bytes()[..].as_ref())?;
+
             let mut prover_file = File::create("pk_testcirc")?;
             prover_file.write(prover_key.to_bytes()[..].as_ref())?;
+
             let mut verifier_file = File::create("vk_testcirc")?;
             verifier_file.write(verifier_key.to_bytes().as_ref())?;
         };
 
-        // Read ProverKey
-        let prover_key = ProverKey::from_bytes(&std::fs::read("pk_testcirc")?[..]).unwrap();
-        // Read VerifierKey
-        let verifier_key = VerifierKey::from_bytes(&std::fs::read("vk_testcirc")?[..]).unwrap();
+        let size = circuit.circuit_size();
 
-        // Generate new inputs
-        // Generate circuit compilation params
-        let inputs2 = [
-            BlsScalar::from(20u64),
-            BlsScalar::from(5u64),
+        // Read the keys
+        let ck = CommitKey::from_bytes(&std::fs::read("ck_testcirc")?[..]).unwrap();
+        let ok = OpeningKey::from_bytes(&std::fs::read("ok_testcirc")?[..]).unwrap();
```

Comment:
I don't think it's worth it to store these two IMO. It's much easier and will perform mostly on the same way to directly provide the PPs always and just trim inside of gadget and compile fn's.

Reply:
The purpose of this PR is not to define the strategy of storing the opening key or commit keys.

This will be implemented by the user, they just need to be fetched and provided to the circuit.

There is no helper function for this serialization / deserialization